distsql/request_builder.go

```diff
@@ -47,6 +47,7 @@ type RequestBuilder struct {
 
 // Build builds a "kv.Request".
 func (builder *RequestBuilder) Build() (*kv.Request, error) {
+	builder.setMatchStoreLabels()
```

Comment:
I think it can be inlined with `builder.verifyTxnScope`. Not a necessary change request.